packages/use-contractkit/src/connectors/useMetaMaskConnector.ts

```diff
@@ -12,16 +12,19 @@ export function useMetaMaskConnector(
     initConnector,
     initError: error,
     dapp,
-  } = useInternalContractKit();
+  } = useContractKitInternal();
 
   useEffect(() => {
     let stale;
     void (async () => {
       const connector = new MetaMaskConnector(network);
-      const { error } = await initConnector(connector);
-      if (error) console.log('got an error');
-      if (!error && !stale) {
-        onSubmit(connector);
+      try {
+        await initConnector(connector);
+        if (!stale) {
```

Comment:
where can stale be assigned a value?

Reply:
Following the `isMounted` logic used elsewhere in this library, I think we need to include a cleanup function in the useEffect that sets stale to true.

Reply:
Line 32 is where stale will be assigned a value. In this case it's different than unmounting. If stale is true that means the values included in the dependency list for this hook have changed somehow while we were `await`ing the `initConnector`

Reply:
Though admittedly this could potentially be problematic if we've initialized the meta mask connector and then we do it again because the `initConnector` function has changed...

Reply:
oh wow i completely failed to see the on unmount function. whoops